src/lib.rs

```diff
@@ -225,13 +225,34 @@ enum SystemState {
 }
 
 /// A marker trait for types which participate in the publish-subscribe system
-/// of the actor framework.
-pub trait Event: Clone + std::any::Any + 'static + Send + Sync {}
+/// of the actor framework. If it makes sense to cache your event and use it with
+/// [`Context::subscribe_and_receive_latest()`], then mark your type with [`CacheableEvent`]
+/// instead.
+pub trait Event: Clone + std::any::Any {
+    /// Internal method for the actor framework: don't implement this downstream. Make your type
+    /// implement [`CacheableEvent`] instead of [`Event`] if it makes sense to cache it.
+    #[doc(hidden)]
+    fn cache(&self) -> Option<Box<dyn std::any::Any + Send + Sync>> {
+        None
+    }
+}
+
+/// A marker trait for types which participate in the publish-subscribe system
+/// of the actor framework and can be cached. [`Context::subscribe_and_receive_latest()`]  will be
+/// available for them in addition to [`Context::subscribe()`]. Blanket implementation of [`Event`]
+/// is provided for all types that implement [`CacheableEvent`].
+pub trait CacheableEvent: Clone + std::any::Any + Send + Sync {}
```

Comment:
I think this should be called `CachedEvent` because if you use it, we don't ask you "would you like to cache this event, please?" - we just cache it.

Reply:
discussed in person: Cacheable is fine!